pkg/util/sys/device.go

```diff
@@ -254,31 +301,46 @@ func UnmountDevice(devicePath string, executor exec.Executor) error {
 	return nil
 }
 
-func DoesDeviceHaveChildren(device string, executor exec.Executor) (bool, error) {
-	cmd := fmt.Sprintf("check children for device %s", device)
-	output, err := executor.ExecuteCommandWithOutput(false, cmd, "lsblk --all -n -l --output PKNAME")
+func GetFSUUID(device string, executor exec.Executor) (string, error) {
+	cmd := fmt.Sprintf("get disk %s fs uuid", device)
+	output, err := executor.ExecuteCommandWithOutput(false, cmd,
+		"udevadm", "info", "--query=property", fmt.Sprintf("/dev/%s", device))
 	if err != nil {
-		return false, fmt.Errorf("command %s failed: %+v", cmd, err)
+		return "", err
 	}
+	return parseFSUUID(output), nil
+}
 
-	searchFor := fmt.Sprintf("^%s$", device)
-	children := Grep(output, searchFor)
+func CheckIfDeviceAvailable(executor exec.Executor, name string) (bool, string, error) {
+	ownPartitions := true
+	partitions, _, err := GetDevicePartitions(name, executor)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to get %s partitions. %+v", name, err)
+	}
+	if !RookOwnsPartitions(partitions) {
+		ownPartitions = false
+	}
 
-	return children != "", nil
+	// check if there is a file system on the device
+	devFS, err := GetDeviceFilesystems(name, executor)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to get device %s filesystem: %+v", name, err)
+	}
+
+	return ownPartitions, devFS, nil
 }
 
-// finds the file system(s) for the device in the output of 'df'
-func parseDFOutput(device, output string) string {
-	var fs []string
-	lines := strings.Split(output, "\n")
-	for _, line := range lines {
-		if strings.HasPrefix(line, fmt.Sprintf("/dev/%s", device)) {
-			words := strings.Split(line, " ")
-			fs = append(fs, words[len(words)-1])
+func RookOwnsPartitions(partitions []*Partition) bool {
+
+	// if there are partitions, they must all have the rook osd label
+	for _, p := range partitions {
+		if !strings.HasPrefix(p.Label, "ROOK-OSD") {
```

Comment:
is `ROOK-OSD` defined in a const somewhere?

Reply:
good catch! the string seems to be everywhere, will do a cleanup in a separate PR.